### PR TITLE
Fix some resolver bugs

### LINKED ISSUE #680: toString does not work in u64 array
Consider the following two functions 
```typescript

export function test(): string {
  let a: u64 = 1;
  return a.toString();
}

export function test1(): string {
  let arr: u64[] = Array.create<u64>(1);
  arr[0] = 1;
  return arr[0].toString();
}
```
The first one compiles but the second one doesn't and the error is 
```
ERROR TS2339: Property 'toString' does not exist on type 'u64'.

   return arr[0].toString();
                 ~~~~~~~~
```
Fiddle example [here](https://webassembly.studio/?f=a304bp7hfwt)

### PR DESCRIPTION
**Pull request by bowenwang1996**
Fix a certain type of bug in the resolver which is caused by primitive types not having `classReference`. More specifically, this PR fixes #680. However, there are some cases, especially those involve operators, are left untouched, so #473 remains open.

### CODE REVIEW

**Comment by MaxGraey** [comment]
Thanks for PR! You should remove (discard) `dist` from update. See [contributing notes]( https://github.com/AssemblyScript/assemblyscript/blob/master/CONTRIBUTING.md#submitting-pull-requests). Also you should add your name to NOTICE file

**Comment by bowenwang1996** [comment]
@MaxGraey done

File: src/resolver.ts [hunk 1]
```diff
@@ -748,14 +748,22 @@ export class Resolver extends DiagnosticEmitter {
...
-            );
-            return null;
+          let classReference = arrayType.classReference;
+          if (!classReference) {
+            let typeClasses = this.program.typeClasses;
+            if (!arrayType.is(TypeFlags.REFERENCE) && typeClasses.has(arrayType.kind)) {
+              classReference = typeClasses.get(arrayType.kind)!;
+            } else {
+              this.error(
+                  DiagnosticCode.Property_0_does_not_exist_on_type_1,
```
Comment (dcodeIO) [inline src/resolver.ts]: nit: indent

**Comment by dcodeIO** [comment]
I think the package-lock differences happen between different environments, and just keeping it as it is helps to reduce some of the noise. Can you reset it?

Other than that, LGTM with the comments resolved :)

File: tests/compiler/resolve-access.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export function arrayAccess(): string {
+    let arr: u64[] = [1];
```
Comment (dcodeIO) [inline tests/compiler/resolve-access.ts]: indentation seems to differ from other tests here too

Comment (bowenwang1996) [reply tests/compiler/resolve-access.ts]: Ah it's because my IDE settings. Will change to 2 spaces

**Comment by dcodeIO** [comment]
Thanks!

**Pull request merged**